libraries/AP_HAL_Linux/UARTDriver.cpp

```diff
@@ -475,8 +377,8 @@ bool UARTDriver::_write_pending_bytes(void)
             n = limit;
         }
     }
-    const uint8_t b = _writebuf[_writebuf_head];
-    if (_packetise && n > 0 &&
+    const int16_t b = _writebuf.peek(0);
+    if (_packetise && n > 0 && b > 0 &&
```

Comment:
Why the `b > 0`? It is already checked that it has to be equal to one of two values.